### PR TITLE
Mapper and repository

### PR DESCRIPTION
**Pull request by Tatiana2424**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by dimasster — APPROVED** [review]

File: Streetcode/Streetcode.BLL/Services/Streetcode/TextContent/FactService.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    public string GetFactsByStreetcode()
+    {
+        Fact fact = new Fact();
+        fact.Id = 1;
+        fact.Title = "Test1";
+        FactDTO factDTO = _mapper.Map<Fact, FactDTO>(fact);
+
+        return factDTO.Title;
+        // return _repositoryWrapper.FactRepository.GetFactsByStreetcode();
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.BLL/Services/Streetcode/TextContent/FactService.cs:28]: Let's add TODO here to clean this after merge

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Media/IAudioRepository.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿
+namespace Repositories.Interfaces;
+
+public interface IAudioRepository
+{
+    public void GetAudioAsync();
+    public void UploadAudioAsync();
+    public void DeleteAudioAsync();
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.DAL/Repositories/Interfaces/Media/IAudioRepository.cs:8]: I think we can remove these duplicated methods from all repository's interfaces and just inherit them from IRepositoryBase<T>

File: Streetcode/Streetcode.DAL/Repositories/Realizations/AdditionalContent/SubtitleRepository.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+public class SubtitleRepository : RepositoryBase<Subtitle>, ISubtitleRepository
+{
+
+    public SubtitleRepository(StreetcodeDbContext _streetcodeDBContext)
+    {
+    }
+
+    public string GetSubtitlesByStreetcode()
+    {
+        return "GetSubtitlesByStreetcode";
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.DAL/Repositories/Realizations/AdditionalContent/SubtitleRepository.cs:18]: Let's add TODO for all code that we will have to clean

File: Streetcode/Streetcode/Controllers/FactController.cs [hunk 4]
```diff
@@ -23,8 +25,10 @@ public void GetFactById()
         // TODO implement here
     }
     [HttpGet("getFactByStreetcode")]
-    public void GetFactsByStreetcode() 
+    public string GetFactsByStreetcode() 
     {
+        _loggerService.LogError("Error????????");
+        return _factService.GetFactsByStreetcode();
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/Controllers/FactController.cs:30]: Let's use Async

File: Streetcode/Streetcode/Extensions/ServiceCollectionExtentions.cs [hunk 5]
```diff
@@ -1,16 +1,33 @@
-﻿using Microsoft.EntityFrameworkCore;
+﻿using EPlast.BLL.Interfaces.Logging;
+using EPlast.BLL.Services.Logging;
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/Extensions/ServiceCollectionExtentions.cs]: Very strange namespace

**Review by LanchevychMaxym — CHANGES_REQUESTED** [review]

File: Streetcode/Streetcode.BLL/Interfaces/Logging/ILoggerService.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.Extensions.Logging;
+
+namespace EPlast.BLL.Interfaces.Logging
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/Interfaces/Logging/ILoggerService.cs]: Copy & Paste code;
Check namespaces everywhere you did that.

File: Streetcode/Streetcode.BLL/Interfaces/Streetcode/TextContent/ITextService.cs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+namespace Streetcode.BLL.Interfaces.Streetcode.TextContent;
+
+public interface ITextService
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/Interfaces/Streetcode/TextContent/ITextService.cs:3]: We don't need some of the repository interfaces;
You can just make its realization dependent on IRepositoryBase.

File: Streetcode/Streetcode.BLL/Interfaces/Streetcode/TextContent/ITermService.cs [hunk 8]
```diff
@@ -0,0 +1,6 @@
+namespace Streetcode.BLL.Interfaces.Streetcode.TextContent;
+
+public interface ITermService
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/Interfaces/Streetcode/TextContent/ITermService.cs:3]: We don't need some of the repository interfaces;
You can just make its realization dependent on IRepositoryBase.

File: Streetcode/Streetcode.BLL/Mapping/AdditionalContent/Coordinates/CoordinateProfile.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+using AutoMapper;
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/Mapping/AdditionalContent/Coordinates/CoordinateProfile.cs:1]: Fix indentation problems in all files from this commit (Unnecessary blank lines, unconventional inheritance spacing, etc).

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs [hunk 10]
```diff
@@ -1,21 +1,21 @@
...
+     void Create();
 
-    public void CreateAsync();
+     Task CreateAsync();
 
-    public void GetById();
+     void GetById();
 
-    public void GetAll();
+     IQueryable<T> GetAll();
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs:13]: When working with IQueryable interface, we are posing a "question", applying all necessary filters right off the bat.
Hence the method's name should be "FindAll" and not "GetAll".

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs [hunk 11]
```diff
@@ -1,21 +1,21 @@
...
 
-    public void GetAllAsync();
+     Task<IEnumerable<T>> GetAllAsync();
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs:15]: Also, it wouldn't hurt to add optional Expressions/Predicates, bound to default/null by default.

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs [hunk 12]
```diff
@@ -1,21 +1,21 @@
 
 namespace StreetCode.DAL.Repositories.Interfaces.Base;
 
-public interface IRepositoryBase
+public interface IRepositoryBase<T>
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs:4]: As a side note, during the latter stages of our development plan we would surely face the need insome form of conditional fetching of only one entity from the repo.
Hence, the base repo interface must be supplied with FirstOrDefaultAsync/SingleOrDefaultAsync methods.

File: Streetcode/Streetcode/Extensions/ServiceCollectionExtentions.cs [hunk 13]
```diff
@@ -1,16 +1,33 @@
...
 using StreetCode.DAL.Repositories.Interfaces.Base;
 
 public static class ServiceCollectionExtentions
 {
     public static IServiceCollection AddAppServices(this IServiceCollection services)
     {
+        services.AddScoped(typeof(ILoggerService<>), typeof(LoggerService<>));
         services.AddScoped<IFactService, FactService>();
         services.AddScoped<IMediaService, MediaService>();
         services.AddScoped<IPartnersService, PartnersService>();
```
Comment (AleXLaeR) [inline Streetcode/Streetcode/Extensions/ServiceCollectionExtentions.cs:33]: Please divide intergration of built-in services and custom ones into their own extension methods.

**Review by AleXLaeR — CHANGES_REQUESTED** [review]
Please thouroughly review my comments & impovement suggestions.

**Review by AleXLaeR — APPROVED** [review]
All done great.

**Review by LanchevychMaxym — APPROVED** [review]
LGTM!

**Pull request merged**